hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestCopyListing.java

```diff
@@ -167,6 +169,77 @@ public void testDuplicates() {
     }
   }
 
+  @Test(expected = DuplicateFileException.class, timeout = 10000)
+  public void testDiffBasedSimpleCopyListing() throws IOException {
+    FileSystem fs = null;
+    Configuration configuration = getConf();
+    DistCpSync distCpSync = Mockito.mock(DistCpSync.class);
+    Path listingFile = new Path("/tmp/list");
+    // Throws DuplicateFileException when copyListing is SimpleCopyListing
+    // as it recursively traverses src3 directory and also adds 3.txt,4.txt twice
+    configuration.set(DistCpConstants.CONF_LABEL_DIFF_COPY_LISTING_CLASS,
+        SimpleCopyListing.class.getName());
```

Comment:
How this config should be configured when we want to copy from HDFS to Ozone or vise versa ?

Reply:
HDFS to Ozone -> traverseDirectory - true as hdfs diff contains only top level dirs
Ozone to HDFS -> traverseDirectory - false as ozone diff contains all subpaths too